packages/react-vapor/src/components/flatSelect/FlatSelectSelectors.ts

```diff
@@ -0,0 +1,16 @@
+import {createSelector} from 'reselect';
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {IFlatSelectState} from './FlatSelectReducers';
+
+const getInput = (state: IReactVaporState, props: {id: string}): IFlatSelectState => {
+    return _.findWhere(state.flatSelect, {id: props.id});
+};
+
+const getSelectedOptionId = createSelector(getInput, (flatSelect: IFlatSelectState): string => flatSelect && flatSelect.selectedOptionId);

```

Comment:
you can remove your check for the flatSelect after that and use that
```suggestion
const getSelectedOptionId = createSelector(getInput, (flatSelect: IFlatSelectState): string => flatSelect.selectedOptionId);
```